plugins/woocommerce/includes/class-wc-shipping.php

```diff
@@ -334,28 +333,36 @@ public function calculate_shipping_for_package( $package = array(), $package_key
 
 		if ( ! is_array( $stored_rates ) || $package_hash !== $stored_rates['package_hash'] || 'yes' === get_option( 'woocommerce_shipping_debug_mode', 'no' ) ) {
 			foreach ( $this->load_shipping_methods( $package ) as $shipping_method ) {
-				if ( ! $shipping_method->supports( 'shipping-zones' ) || $shipping_method->get_instance_id() ) {
-					/**
-					 * Fires before getting shipping rates for a package.
-					 *
-					 * @since 4.3.0
-					 * @param array $package Package of cart items.
-					 * @param WC_Shipping_Method $shipping_method Shipping method instance.
-					 */
-					do_action( 'woocommerce_before_get_rates_for_package', $package, $shipping_method );
-
-					// Use + instead of array_merge to maintain numeric keys.
-					$package['rates'] = $package['rates'] + $shipping_method->get_rates_for_package( $package );
-
-					/**
-					 * Fires after getting shipping rates for a package.
-					 *
-					 * @since 4.3.0
-					 * @param array $package Package of cart items.
-					 * @param WC_Shipping_Method $shipping_method Shipping method instance.
-					 */
-					do_action( 'woocommerce_after_get_rates_for_package', $package, $shipping_method );
+				// If the shipping method supports shipping zones and is not an instance, skip it.
+				if ( $shipping_method->supports( 'shipping-zones' ) && ! $shipping_method->get_instance_id() ) {
+					continue;
+				}
+
+				// If the package is not shippable and the shipping method does not support local pickup, skip it.
+				if ( ! $is_shippable && ! $shipping_method->supports( 'local-pickup' ) ) {
```

Comment:
This line allows local pickup rates to be returned even if the "shipping country" is invalid according to settings.